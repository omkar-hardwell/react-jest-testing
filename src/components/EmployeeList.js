import React, { useEffect, useState } from "react";

import { getEmployeeList } from "../Api/dummyAPIs";

const EmployeeList = () => {
  const [employeeList, setEmployeeList] = React.useState(null);
  useEffect(() => {
    fetchDetails();
  }, []);

  const fetchDetails = () => {
    getEmployeeList().then((res) => setEmployeeList(res));
  };

  return (
    <>
      <h1>Employee Details</h1>
      <hr />
      {employeeList && employeeList.employee.length > 0 ? (
        employeeList.employee.map((employee) => {
          return (
            <React.Fragment key={employee.id}>
              <div>
                <b>Employee ID :</b> {employee.id}
              </div>
              <div>
                <b>Employee Name :</b> {employee.name}
              </div>
              <div>
                <b>Designation :</b> {employee.designation}
              </div>
              <div>
                <b>Department :</b> {employee.department}
              </div>
              <hr />
            </React.Fragment>
          );
        })
      ) : (
        <p>No records founds!</p>
      )}
    </>
  );
};

export default EmployeeList;
