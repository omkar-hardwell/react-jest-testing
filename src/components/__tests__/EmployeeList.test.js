// import { render, screen } from "@testing-library/react";
import React from "react";
import { shallowToJson } from "enzyme-to-json";
import { shallow } from "enzyme";

import EmployeeList from "../EmployeeList";

import "../../setupTests.js";

describe("Test EmployeeList component with", () => {
  let wrapper;

  const dummyEmployeeList = {
    employee: [
      {
        id: 1,
        name: "Omkar",
        designation: "React developer",
        department: "UI",
      },
      {
        id: 2,
        name: "Jigar",
        designation: "React developer",
        department: "UI",
      },
    ],
  };
  const employeeListNotFound = null;
  const employeeListWithZeroData = { employee: [] };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("data found", () => {
    /**
     * Without snapshot, simply redering a component
     */
    // render(<EmployeeList employeeList={dummyEmployeeList} />);
    /**
     * Snapshot testing
     */
    React.useState = jest.fn((initialValue) => [dummyEmployeeList, () => {}]);
    wrapper = shallow(<EmployeeList />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it("data not defined", () => {
    React.useState = jest.fn((initialValue) => [
      employeeListNotFound,
      () => {},
    ]);
    wrapper = shallow(<EmployeeList />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it("Zero data found", () => {
    React.useState = jest.fn((initialValue) => [
      employeeListWithZeroData,
      () => {},
    ]);
    wrapper = shallow(<EmployeeList />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
