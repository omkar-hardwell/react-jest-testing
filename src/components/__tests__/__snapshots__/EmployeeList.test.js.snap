// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test EmployeeList component with Zero data found 1`] = `
<Fragment>
  <h1>
    Employee Details
  </h1>
  <hr />
  <p>
    No records founds!
  </p>
</Fragment>
`;

exports[`Test EmployeeList component with data found 1`] = `
<Fragment>
  <h1>
    Employee Details
  </h1>
  <hr />
  <div>
    <b>
      Employee ID :
    </b>
     
    1
  </div>
  <div>
    <b>
      Employee Name :
    </b>
     
    Omkar
  </div>
  <div>
    <b>
      Designation :
    </b>
     
    React developer
  </div>
  <div>
    <b>
      Department :
    </b>
     
    UI
  </div>
  <hr />
  <div>
    <b>
      Employee ID :
    </b>
     
    2
  </div>
  <div>
    <b>
      Employee Name :
    </b>
     
    Jigar
  </div>
  <div>
    <b>
      Designation :
    </b>
     
    React developer
  </div>
  <div>
    <b>
      Department :
    </b>
     
    UI
  </div>
  <hr />
</Fragment>
`;

exports[`Test EmployeeList component with data not defined 1`] = `
<Fragment>
  <h1>
    Employee Details
  </h1>
  <hr />
  <p>
    No records founds!
  </p>
</Fragment>
`;
